src/data-workspace/use-data-value-mutation/use-mutation-error.js

```diff
@@ -0,0 +1,28 @@
+import { useAlert } from '@dhis2/app-runtime'
+
+const useMutationError = () => {
+    const { show: showAlert } = useAlert(
+        ({ message }) => message,
+        () => ({
+            critical: true,
+        })
+    )
+
+    /**
+     * dictionary of error codes we want to handle by showing the server-side message, any error code not in the dictionary will not be handled
+     */
+    const errors = {
+        E2017: true /** E2017 dataset is locked https://github.com/dhis2/dhis2-core/blob/28a6aa052f221626e583a84bc18d80c0a8fa0927/dhis-2/dhis-api/src/main/java/org/hisp/dhis/feedback/ErrorCode.java#L97 */,
+    }
+
+    const onError = (error) => {
+        if (errors[error?.details?.errorCode]) {
```

Comment:
@Mohammer5 @HendrikThePendric .. in this PR, I am working on handling the 409 (dataset is locked) scenario. Do we only want to show the server-side error message in this case, or do we want to show the message sent back from the server for any error (or maybe any 4xx error)